core/src/banking_stage/transaction_scheduler/scheduler_controller.rs

```diff
@@ -655,6 +683,34 @@ impl<T: LikeClusterInfo> SchedulerController<T> {
     }
 }
 
+/// Given the last slot in the epoch, the minimum deactivation slot,
+/// and the current slot, return the `MaxAge` that should be used for
+/// the transaction. This is used to determine the maximum slot that a
+/// transaction will be considered valid for, without re-resolving addresses
+/// or resanitizing.
+///
+/// This function considers the deactivation period of Address Table
+/// accounts. If the deactivation period runs past the end of the epoch,
+/// then the transaction is considered valid until the end of the epoch.
+/// Otherwise, the transaction is considered valid until the deactivation
+/// period.
+///
+/// Since the deactivation period technically uses blocks rather than
+/// slots, the value used here is the lower-bound on the deactivation
+/// period, i.e. the transaction's address lookups are valid until
+/// AT LEAST this slot.
+fn calculate_max_age(
+    last_slot_in_epoch: Slot,
+    deactivation_slot: Slot,
+    current_slot: Slot,
+) -> MaxAge {
+    let alt_min_expire_slot = estimate_last_valid_slot(deactivation_slot.min(current_slot));
```

Comment:
Technically this should be `current_slot + 1` I think because we know that the ALT didn't get deactivated in the `current_slot`